addon/index.js

```diff
@@ -17,43 +18,44 @@ export default Mixin.create({
     this._super(...arguments);
 
     setupRouter(this);
-  },
 
-  destroy() {
-    reset();
+    if (gte('3.6.0-beta.1')) {
+      this.on('routeWillChange', () => {
+        this._routeWillChange();
+      });
 
-    this._super(...arguments);
-  },
+      this.on('routeDidChange', (transition) => {
+        this._routeDidChange(transition);
+      });
+    } else {
+      this.willTransition = (...args) => {
+        this._super(...args);
```

Comment:
I don't think `this._super` will work in this context (because this arrow function won't be super wrapped which happens during `.extend`).

Reply:
I think generally speaking we want to restructure to something like:

```js
let RouterScrollMixin = Mixin.create({
  // shared code, and new `init` (with `router.on('routeWillChange'` / `router.on('routeDidChange'` behind the gte guard)
});

if (!gte('3.6.0-beta.1')) {
  RouterScrollMixin = Mixin.create(RouterScrollMixin, {
    willTransition() {},
    didTransition() {},
  });
}

export default RouterScrollMixin;
```

Reply:
Makes sense, will update